src/Integration/UnintrusiveBinding/UnintrusiveConfigurationProvider.cs

```diff
@@ -19,15 +19,54 @@
  */
 
 using System.ComponentModel.Composition;
+using System.IO;
 using SonarLint.VisualStudio.Core.Binding;
+using SonarLint.VisualStudio.Integration.Persistence;
+using SonarLint.VisualStudio.Integration.UnintrusiveBinding;
 
 namespace SonarLint.VisualStudio.Integration.NewConnectedMode
 {
     [Export(typeof(IConfigurationProvider))]
     [PartCreationPolicy(CreationPolicy.Shared)]
     internal class UnintrusiveConfigurationProvider : IConfigurationProvider
     {
-        // TODO - fetch settings from new location #4170
-        public BindingConfiguration GetConfiguration() => BindingConfiguration.Standalone;
+        private readonly IUnintrusiveBindingPathProvider pathProvider;
+        private readonly ISolutionBindingDataReader solutionBindingDataReader;
+
+        [ImportingConstructor]
+        public UnintrusiveConfigurationProvider(
```

Comment:
This is very similar to the old [ObsoleteConfigurationProvider](https://github.com/SonarSource/sonarlint-visualstudio/blob/173efdcd16e3af13cbf6394bc129bc81ce634af7/src/Integration/NewConnectedMode/ObsoleteConfigurationProvider.cs#L52), but simpler.
The old class handled two different locations for Connected Mode settings. This version only needs to handle one.